Add Contributor's Guide (#13)
ProjectPythiaTutorials/cookbook-template: CONTRIBUTING.md

# Project Pythia Cookbook Contributor's Guide

Project Pythia Cookbooks are collections of more advanced and domain-specific example
workflows building on top of [Pythia Foundations](https://foundations.projectpythia.org/landing-page.html). 
They are [geoscience](https://en.wikipedia.org/wiki/Earth_science)-focused

> and should direct the reader towards the Foundations material for any required

Author: Done

background knowledge. 

The following is a step-by-step guide to getting your cookbook idea
hosted on the Project Pythia website.

1. Use the template
    1. On https://github.com/ProjectPythiaTutorials/cookbook-template, click "Use this template"
    1. Create the repo under your account with a descriptive name, followed by `-cookbook` (e.g. `hydrology-cookbook`, `hpc-cookbook`, `cesm-cookbook`, etc.)
    1. See the [GitHub tutorial in Foundations](https://foundations.projectpythia.org/foundations/getting-started-github.html) for how to set up a clone
1. Create the environment

>     1. Edit `environment.yml`: change the name to `<your-cookbook-name>-dev` and add all required libraries and other dependencies under `dependencies:`. Commit the changes

Author: Done

    1. Create the Conda environment with `conda env create -f environment.yml`. If it crashes, try running `conda config --set channel_priority strict`
    1. Activate your environment with `conda activate <env-name>`
1. Add content
    1. Using the notebook template, add your content. Add folders to organize notebooks into sections if applicable
    1. Add the notebooks to `_toc.yml`. See `radar-cookbook/notebooks/_toc.yml` for syntax

>     1. Change `README.md` to include your cookbook title, various badges, a sentence or two describing the cookbook, and a link to the landing page. See the [Radar Cookbook](https://github.com/ProjectPythiaTutorials/radar-cookbook/blob/main/README.md) for formatting

Reviewer: Can you change "formatting" to "an example"?

Author: Done

> 1. Transfer cookbook to [ProjectPythiaTutorials](https://github.com/ProjectPythiaTutorials)

Author: Done

# Project Pythia Cookbook Contributor's Guide

Project Pythia Cookbooks are collections of more advanced and domain-specific example
workflows building on top of [Pythia Foundations](https://foundations.projectpythia.org/landing-page.html). 
They are [geoscience](https://en.wikipedia.org/wiki/Earth_science)-focused
and should direct the reader towards the [Foundations material](https://foundations.projectpythia.org/landing-page.html) for any required
background knowledge. 

The following is a step-by-step guide to getting your cookbook idea
hosted on the [Project Pythia Cookbooks gallery](https://projectpythia.org/cookbook-gallery.html).

1. Use the template

>     1. On https://github.com/ProjectPythiaTutorials/cookbook-template, click "Use this template"

Reviewer 2: ```suggestion
    1. If you don't already have a GitHub account, create one by following the Getting Started with GitHub guide [here](https://foundations.projectpythia.org/foundations/getting-started-github.html)
    1. On https://github.com/ProjectPythiaTutorials/cookbook-template, click "Use this template"
```

>     1. Create the repo under your account with a descriptive name, followed by `-cookbook` (e.g. `hydrology-cookbook`, `hpc-cookbook`, `cesm-cookbook`, etc.)

Reviewer 2: ```suggestion
    1. Create the repo under your account with a descriptive name, followed by `-cookbook` (e.g. `hydrology-cookbook`, `hpc-cookbook`, `cesm-cookbook`, etc.) by entering a name into the `Repository name` field and clicking on `Create repository from template`.
```

>     1. In the settings, enable GitHub Pages and make the source `gh-pages/(root)`

Reviewer 2: ```suggestion
    1. Your browser should be directed to the newly created repository under your GitHub account. Under Settings, enable GitHub Pages by changing the Source from `none` to `main` and clicking on `Save`
```

Author: Done, but `main` should be `gh-pages`

Reviewer 2: Can you confirm this? I'm pretty sure when I went through the process the pull-down gave me the option of selecting 'none' or 'main'. Thanks.

Author: Now that I look at it, there's an option "Include all branches" when using the template that would need to be chosen to get the `gh-pages` branch as well. I'll add that to the guide.

>     1. See the [GitHub tutorial in Foundations](https://foundations.projectpythia.org/foundations/getting-started-github.html) for how to set up a clone

Reviewer 2: A little more explicit guidance needed here.

Author: Added "Clone the repo in your local workspace and `cd` into your cookbook directory"

1. Create the environment

>     1. Edit `environment.yml`: change the name to `<your-cookbook-name>-dev` (e.g. `cesm-cookbook-dev`) and add all required libraries and other dependencies under `dependencies:`. Commit the changes

Reviewer 2: Is it essential that the `name` here is identical to the name set in step 1.i above?

Author: No, but I think it is probably a good idea for consistency

    1. Create the Conda environment with `conda env create -f environment.yml`. If it crashes, try running `conda config --set channel_priority strict`
    1. Activate your environment with `conda activate <env-name>`
1. Add content

>     1. Using the notebook template, add your content. Add folders to organize notebooks into sections if applicable

Reviewer 2: Does this mean edit the file `notebooks/notebook-template.ipynb` directly to add content?

Author: Yes, added clarifying line

>     1. Using the notebook template, add your content. Add folders to organize notebooks into sections if applicable

Reviewer 2: ```suggestion
    1. Create a new git branch to work in. See [here](https://foundations.projectpythia.org/foundations/github/git-branches.html) for reference
    1. Using the notebook template, add your content. Add folders to organize notebooks into sections if applicable
```

    1. Add the notebooks to `_toc.yml`. See `radar-cookbook/notebooks/_toc.yml` for syntax
    1. Change `README.md` to include your cookbook title, various badges, a sentence or two describing the cookbook, and a link to the landing page. See the [Radar Cookbook](https://github.com/ProjectPythiaTutorials/radar-cookbook/blob/main/README.md) for an example

>     1. When you open a PR on your cookbook repo, the github-actions bot will comment a link to a preview of your cookbook

Reviewer 2: ```suggestion
    1. Commit your changes with git, and create a PR to your repo. See [here](https://foundations.projectpythia.org/foundations/github/github-pull-request.html).
    1. When you open a PR on your cookbook repo, the github-actions bot will comment a link to a preview of your cookbook
```

Author: Done
